components/style/matching.rs

```diff
@@ -662,13 +662,9 @@ pub trait MatchMethods : TNode {
 
         let damage;
         if self.is_text_node() {
-            // Text nodes get a copy of the parent style. This ensures
-            // that during fragment construction any non-inherited
-            // CSS properties (such as vertical-align) are correctly
-            // set on the fragment(s).
             let mut data_ref = self.mutate_data().unwrap();
             let mut data = &mut *data_ref;
-            let cloned_parent_style = parent_style.unwrap().clone();
+            let cloned_parent_style = Arc::new(parent_style.unwrap().clone_for_text_node());
```

Comment:
So, the problem with this (for the servo case) is that it turns an Arc<T>::clone into a T::clone, which means that it's no longer a refcount bump but instead an actual copy of the ComputedValues. This is still relatively cheap compared to cloning actual style data - and indeed I hope it doesn't matter, since we need to do it for Gecko. But I still think we shouldn't introduce anything that might make Servo slower.

So let's change clone_for_text node to be a static method of the form &Arc<Self> -> Arc<Self>. That will let the servo implementation just return a clone of the arg, and the gecko implementation can do what it does.

And while we're at it, I would rename it to "style_for_child_text_node", which is a bit clearer.